### PR TITLE
Feature/version 1.1.0

### PR DESCRIPTION
**Pull request by firas-sitech**
code fix

### CODE REVIEW

File: src/main/java/me/sitech/apifort/api/CacheRest.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        onException(Exception.class).handled(true).process(exception).marshal().json();
+
+        rest("/admin-api/cache/")
+                .description("APIFort Cache Endpoints")
+                .tag("APIFort Cache")
+
+            .delete("/{cache_key}")
+                .id(ApiFortCamelRestIds.REST_DELETE_ITEM_CACHE_ROUTE_ID)
+                .description("Delete cache data by key")
+                .responseMessage().code(ApiFortStatusCode.UNAUTHORIZED).message(ApiFortStatusCode.UNAUTHORIZED_STRING).responseModel(GeneralRes.class).endResponseMessage()
```
Comment (iabughosh) [inline src/main/java/me/sitech/apifort/api/CacheRest.java:40]: Make sure layout is fine.

File: src/main/java/me/sitech/apifort/api/HealthRest.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+public class HealthRest extends RouteBuilder {
+
+    private final ExceptionHandlerProcessor exception;
+
+    @Inject
+    public HealthRest(ExceptionHandlerProcessor exception){
+        this.exception=exception;
+    }
+
+    @Override
```
Comment (iabughosh) [inline src/main/java/me/sitech/apifort/api/HealthRest.java:26]: Why this is created? we already have microprofile health extension.

File: src/main/java/me/sitech/apifort/domain/module/ProfileCounts.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+    @Column(name = "service_count")
+    private Long serviceCount;
+
+    @Column(name = "endpoint_count")
+    private Long endpointCount;
+
+
+    public ProfileCounts map(Object[] obj){
+        this.uuid = (String) obj[0];
```
Comment (iabughosh) [inline src/main/java/me/sitech/apifort/domain/module/ProfileCounts.java:24]: Do some validations on the argument like null checks and array is not empty.

File: src/main/java/me/sitech/apifort/domain/response/profile/ClientProfileDetailsRes.java [hunk 4]
```diff
@@ -13,15 +13,27 @@ public class ClientProfileDetailsRes {
...
+    private String title;
+
+    @JsonProperty("description")
+    private String description;
+
+    @JsonProperty("api_key")
     private String apiKey;
 
-    @JsonProperty("client_public_certificate")
+    @JsonProperty("public_certificate")
```
Comment (iabughosh) [inline src/main/java/me/sitech/apifort/domain/response/profile/ClientProfileDetailsRes.java:25]: To be confirmed if this is still backward compatible?!

File: src/main/java/me/sitech/apifort/router/v1/client_profile/ClientProfileRouter.java [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+
+        onException(Exception.class).handled(true).process(exception).marshal().json();
+
+        from(DIRECT_GET_CLIENT_PROFILE_ROUTE)
+                .routeId(DIRECT_GET_CLIENT_PROFILE_ROUTE_ID)
+                .to(JwtAuthenticationRoute.DIRECT_JWT_AUTH_ROUTE)
+                .process(exchange -> {
+                    String apiKey = exchange.getIn().getHeader(ApiFort.API_KEY_HEADER, String.class);
+                    log.info("Request API key is {}", apiKey);
+                    if(Util.isNotEmpty(apiKey)){
```
Comment (iabughosh) [inline src/main/java/me/sitech/apifort/router/v1/client_profile/ClientProfileRouter.java:87]: I would prefer to use early-exit pattern here on apiKey validation..

File: src/main/java/me/sitech/apifort/router/v1/client_profile/ClientProfileRouter.java [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+                        return;
+                    }
+                    throw new APIFortGeneralException("Missing api-key");
+                }).marshal().json();
+
+        from(DIRECT_GET_CLIENT_PROFILE_BY_REALM_ROUTE)
+                .routeId(DIRECT_GET_CLIENT_PROFILE_BY_REALM_ROUTE_ID)
+                .to(JwtAuthenticationRoute.DIRECT_JWT_AUTH_ROUTE)
+                .process(exchange -> {
+                    String realm = exchange.getIn().getHeader(ApiFort.API_REALM, String.class);
```
Comment (iabughosh) [inline src/main/java/me/sitech/apifort/router/v1/client_profile/ClientProfileRouter.java:107]: Same

File: src/main/java/me/sitech/apifort/router/v1/gateway/GatewayRouter.java [hunk 7]
```diff
@@ -100,6 +99,7 @@ public void configure() {
                 .log(LoggingLevel.INFO,DOWNSTREAM_ENDPOINT_HEADER)
                 .setHeader(Exchange.HTTP_METHOD, constant(APPLICATION_PATCH))
                 .toD(String.format(CAMEL_BRIDGE_ROUTING_PATH, DOWNSTREAM_ENDPOINT_HEADER))
+
```
Comment (iabughosh) [inline src/main/java/me/sitech/apifort/router/v1/gateway/GatewayRouter.java:102]: Remove extra line,

File: src/main/resources/application.properties [hunk 8]
```diff
@@ -52,8 +51,8 @@ quarkus.generate-code.grpc.scan-for-imports=all
 quarkus.flyway.clean-at-start=false
 quarkus.flyway.migrate-at-start=true
 quarkus.flyway.baseline-on-migrate=true
-#quarkus.flyway.locations=db/migration/ddl
-quarkus.flyway.locations=db/migration/ddl,db/migration/delmonte
+quarkus.flyway.locations=db/migration/ddl
+#quarkus.flyway.locations=db/migration/ddl,db/migration/delmonte
```
Comment (iabughosh) [inline src/main/resources/application.properties:55]: Remove comment on this please..

File: src/main/resources/db/migration/ddl/V1.0.1__UPDATE__api-fort.sql [hunk 9]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE `apifort_client_profile` ADD COLUMN `title` VARCHAR(100) NULL AFTER `uuid`;
```
Comment (iabughosh) [inline src/main/resources/db/migration/ddl/V1.0.1__UPDATE__api-fort.sql:1]: No man please, for any DB change please create new file. db migration is an append only directory. Modifying an existing scripts file, an existing projects will not be able to start unless you cleared the out the whole schema.

**Review by iabughosh — CHANGES_REQUESTED** [review]

File: src/main/java/me/sitech/apifort/api/HealthRest.java (hunk 2 above)
Comment (firas-sitech) [reply src/main/java/me/sitech/apifort/api/HealthRest.java:26]: for check the token it is working fine

File: src/main/java/me/sitech/apifort/api/CacheRest.java (hunk 1 above)
Comment (firas-sitech) [reply src/main/java/me/sitech/apifort/api/CacheRest.java:37]: this is additional api we could remove it if not need it

File: src/main/java/me/sitech/apifort/router/v1/client_profile/ClientProfileRouter.java (hunk 5 above)
Comment (firas-sitech) [reply src/main/java/me/sitech/apifort/router/v1/client_profile/ClientProfileRouter.java:87]: what do you mean?

File: src/main/java/me/sitech/apifort/router/v1/gateway/GatewayRouter.java (hunk 7 above)
Comment (firas-sitech) [reply src/main/java/me/sitech/apifort/router/v1/gateway/GatewayRouter.java:102]: which line?

**Pull request closed without merging**